File: 2021-22/contest/Editorial/Codeshows Coding Contest 1/DisableAlarm.cpp
#include<bits/stdc++.h>
using namespace std;

int main()
{
	long long int n,t,x;
	cin>>n>>x>>t;
	long long int arr[n];
	for(long long int i=0;i<n;i++)
	{
		cin>>arr[i];
	}
	 long long int i=0,j=0,sum=arr[0],ans=n+1;
        while(i<=j && j<n)
        {
            if(sum>x)
            {
                sum-=arr[i];
                ans=min(ans,j-i+1);
                i++;
            }
            else
            {
				j++;
				if(j<n)
				{
				sum+=arr[j];
				}
            }
        }
	    cout<<ans*t<<endl;
}

File: 2021-22/contest/Editorial/Codeshows Coding Contest 1/DisableAlarm.py
n,x,t=map(int,input().split())
l=list(map(int,input().split()))
i=0
j=0
ans=n+1
su=l[0]
while i<=j and j<n:
    if su>x:
        su=su-l[i]
        ans=min(ans,j-i+1)
        i=i+1
    else:
        j=j+1
        if j<n:
            su=su+l[i]
print(ans*t)                

